src/mlpack/core/util/cli.cpp

```diff
@@ -400,15 +320,42 @@ void CLI::ParseCommandLine(int argc, char** line)
   po::variables_map& vmap = GetSingleton().vmap;
   po::options_description& desc = GetSingleton().desc;
 
-  // Parse the command line, place the options & values into vmap
+  // Parse the command line, place the options & values into vmap.
   try
   {
-    // Get the basic_parsed_options
+    // Get the basic_parsed_options.
     po::basic_parsed_options<char> bpo(
-      po::parse_command_line(argc, line, desc));
+        po::parse_command_line(argc, line, desc));
 
-    // Look for any duplicate parameters, removing duplicate flags
-    RemoveDuplicateFlags(bpo);
+    // Iterate over all the program_options, looking for duplicate parameters.
+    // If we find any, remove the duplicates.
+    for (unsigned int i = 0; i < bpo.options.size(); i++)
+    {
+      for (unsigned int j = i + 1; j < bpo.options.size(); j++)
+      {
+        if (bpo.options[i].string_key == bpo.options[j].string_key)
+        {
+          // If a duplicate is found, check to see if either one has a value.
+          if (bpo.options[i].value.size() == 0 &&
+              bpo.options[j].value.size() == 0)
+          {
+            // If neither has a value, consider it a duplicate flag and remove
+            // the duplicate. It's important to not break out of this loop
+            // because there might be another duplicate later on in the vector.
+            bpo.options.erase(bpo.options.begin() +  j);
```

Comment:
I think, If we don't break, and we remove another duplicate later on in the vector, we could run into an index out of bounds error.